src/grapple.c

```diff
@@ -327,37 +366,47 @@ void GrappleService()
 		VectorSubtract(self->hook->s.v.origin, self->s.v.origin, hookVector);
 
 	// No longer going to factor maxspeed into grapple velocity
-	// Using standard grapple velocity of 800 set from original 3wave
 	// purectf velocity = 2.35 * 320 or 360 * 1 = 750 or 846
 
 	VectorCopy(hookVector, hookVelocity);
 	VectorNormalize(hookVelocity);
 
+	if (cvar("k_ctf_cr_hook"))
+	{
+		if (self->grapple_vel_increment > 0) // accelerate
+		{
+			self->grapple_vel = IncrementSpeed(self->grapple_vel, self->grapple_vel_increment);
+		}
+		else if (self->grapple_vel_increment < 0) // decelerate
+		{
+			self->grapple_vel = DecrementSpeed(self->grapple_vel, self->grapple_vel_increment);
+		}
+	}
+	else
+	{
+		self->grapple_vel = 800;
+	}
+
+
 	if (self->ctf_flag & CTF_RUNE_HST)
 	{
-		VectorScale(hookVelocity, 800 + (100 * (cvar("k_ctf_rune_power_hst") / 2 + 1)),
+		VectorScale(hookVelocity, self->grapple_vel * 1.125,
```

Comment:
This takes a step away from traditional haste settings. PureCTF lineage of CTF defaulted to hst power 1.25, thus player moved at maxspeed * 1.25, as well as hook movement * 1.25, and shooting speed * 1.25.